spec/rubocop/cop/rspec/file_path_spec.rb

```diff
@@ -232,7 +232,7 @@ module Medium
   end
 
   it 'does not register offense for absolute file path' do
-    allow(File).to receive(:expand_path).with('my_class_spec.rb').and_return(
+    allow(cop).to receive(:expanded_file_path).and_return(
```

Comment:
Stubject detected!

Wondering why this works, while the other one doesn't :thinking:  as [`expanded_file_path` just delegates to `File.expand_path`](https://github.com/rubocop/rubocop-rspec/blob/a46bc5e9688d60b8c0e4317cdd09af634aa85a22/lib/rubocop/cop/rspec/file_path.rb#L167).

Reply:
Since rubocop 1.40.0, `File.expand_path` is always called from `RuboCop::Cop::Base#file_name_matches_any?`, so stubbing this in a test case will fail because a different argument is passed there.

- https://github.com/rubocop/rubocop/commit/54bd547fb9b2b457023e9cbb3b3cde185a62a427#diff-7c7ad2faad81954938dc1293695ac6c6a8e6f0f87a87526f411f6c6f0fbdd767L415-R416

On the other hand, stubbing `cop.expanded_file_path` does not pose a problem here.